Watchman.Discord/Areas/Administration/Services/SetRoleService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Devscord.DiscordFramework.Framework.Commands.Responses;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using MongoDB.Driver;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.DomainModel.DiscordServer.Commands;
+using Watchman.DomainModel.DiscordServer.Queries;
+
+namespace Watchman.Discord.Areas.Administration.Services
+{
+    public class SetRoleService
+    {
+        private readonly IQueryBus _queryBus;
+        private readonly ICommandBus _commandBus;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly UsersRolesService _usersRolesService;
+        public SetRoleService(MessagesServiceFactory messagesServiceFactory, IQueryBus queryBus, ICommandBus commandBus, UsersRolesService usersRolesService)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._queryBus = queryBus;
+            this._usersRolesService = usersRolesService;
+            this._commandBus = commandBus;
+        }
+
+        public async Task SetRolesAsSafe(Contexts contexts, IEnumerable<string> commandRoles, bool setAsSafe)
+        {
+            var safeRolesQuery = new GetDiscordServerSafeRolesQuery(contexts.Server.Id);
+            var safeRoles = this._queryBus.Execute(safeRolesQuery).SafeRoles;
+            var messageService = _messagesServiceFactory.Create(contexts);
+            foreach (var roleName in commandRoles)
+            {
+                var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
+
+                if (serverRole == null)
+                {
+                    await messageService.SendResponse(x => x.RoleNotFoundOrIsNotSafe(contexts, roleName), contexts);
+                    continue;
+                }
+
+                if (setAsSafe && !safeRoles.Select(x => x.Name).Where(x => x == roleName).Select(x => x).Any())
```

Comment:
jest ok, ale zrobiłym tu metody dla lepszej czytelności
bo sporo ifów sie tu zrobiło 
```csharp
if(setAsSafe)
{
    this.TrySetAsSafe() //w środku sprawdzanie tego warunku i później await _commandBus.ExecuteAsync(new SetRoleAsSafeCommand(roleName, contexts.Server.Id));
}
else
{
    this.TrySetAsUnsafe()
}
```
metody prywatne
oczywiście - przekazuj im odpowiednie argumenty

Reply:
czy ty tu robisz selecta
tylko po to żeby zrobić where
tylko po to żeby zrobić selecta
tylko po to żeby zrobić any 

czemu nie zrobisz tego samym `Any(x => x.Name == roleName)`?